cpp/src/arrow/json/converter_test.cc

```diff
@@ -96,5 +96,27 @@ TEST(ConverterTest, Timestamp) {
   AssertConvert(timestamp(TimeUnit::SECOND), src, src);
 }
 
+TEST(ConverterTest, Decimal128) {
+  std::string src = R"([
+          "02.0000000000",
+          "30.0000000000",
+          "22.0000000000",
+        "-121.0000000000",
+        null])";
+
+  AssertConvert(decimal128(38, 10), src, src);
```

Comment:
How does this verify that the value was converted to the correct value if we aren't passing in a decimal128 with the correct value specified?

Reply:
Namely that I would expect something more like instantiating decimals of the correct precision with the values you specified there and then verifying that they are indeed the same values in the same order.

Reply:
The test is confusing because we use the same string representation to produce the expected one. I think that rewriting this test should be clear such confusion.

Reply:
@felipeblazing I just did a refactoring of the tests, and now I think the intention is much clearer. So now, the expected and input values are separate instances, both in JSON format. Namely, the first is an array from JSON that holds the expected values, and the second one is a one object per line JSON table as the input string (thank you @bkietz for your support here).